src/Language/PureScript/AST/Exported.hs

```diff
@@ -144,3 +141,16 @@ isDctorExported ident (Just exps) ctor = test `any` exps
   test (TypeRef _ ident' Nothing) = ident == ident'
   test (TypeRef _ ident' (Just ctors)) = ident == ident' && ctor `elem` ctors
   test _ = False
+
```

Comment:
I suppose we could make a note of the original export lists before desugaring so that we don't have to make any changes to Sugar.Names, but I think that would be a bit awkward. Also I think a good reason to have it there is that other places can benefit, e.g. I think :browse in the repl currently uses an ordering which doesn't resemble either the decl ordering or the export list ordering, which is not ideal.